src/index.js

```diff
@@ -61,126 +66,197 @@ const monitorChromecastPlayer = function (player, options) {
+                postUrl = event.requestData.media.metadata.images[0].url;
+              }
             }
-            if (event.requestData.media.metadata.images !== undefined && event.requestData.media.metadata.images.length > 0) {
-              postUrl = event.requestData.media.metadata.images[0].url;
+
+            if (event.requestData.media.breakClips !== undefined) {
+              adBreakClips = event.requestData.media.breakClips;
             }
           }
-        }
-        if (event.requestData.autoplay !== undefined) {
-          autoplay = event.requestData.autoplay;
-        }
-
-        if (firstPlay) {
-          firstPlay = false;
-        } else {
-          player.mux.emit('videochange', { video_title: title });
-          videoChanged = true;
-          player.mux.emit('ended');
-        }
-        player.mux.emit('loadstart');
-        player.mux.emit('play');
-        break;
-      case cast.framework.events.EventType.MEDIA_FINISHED:
-      case cast.framework.events.EventType.LIVE_ENDED:
-        if (!videoChanged) {
-          player.mux.emit('ended');
-        }
-        videoChanged = false;
-        break;
-      case cast.framework.events.EventType.MEDIA_STATUS:
-        if (event.mediaStatus.videoInfo !== undefined) {
-          // Note: it appears the videoInfo field is always undefined
-          videoSourceWidth = event.mediaStatus.videoInfo.width;
-          videoSourceHeight = event.mediaStatus.videoInfo.height;
-        }
-        if (event.mediaStatus.media !== undefined &&
-          event.mediaStatus.media.duration !== undefined) {
-          duration = mux.utils.secondsToMs(event.mediaStatus.media.duration);
-        }
-        break;
-      case cast.framework.events.EventType.SEEKING:
-        isSeeking = true;
-        player.mux.emit('seeking');
-        break;
-      case cast.framework.events.EventType.SEEKED:
-        player.mux.emit('seeked');
-        break;
-      case cast.framework.events.EventType.PAUSE:
-        isPaused = true;
-        if (!isSeeking) {
-          // Note:
-          // When a seek happens, cast sdk sends the `cast.framework.events.EventType.PAUSE`.
-          // We need to suppress the player.mux.emit('pause') call, because if it goes out, in our data view, seek
-          // events will become pause event
-          player.mux.emit('pause');
-        }
-        break;
-      case cast.framework.events.EventType.PLAYING:
-        if (isPaused) {
+          if (event.requestData.autoplay !== undefined) {
+            autoplay = event.requestData.autoplay;
+          }
+
+          // the user is expecting us to detect video changes
+          if (options.automaticVideoChange && !firstPlay) {
+            player.mux.emit('ended');
+            player.mux.emit('videochange', { video_title: title });
+            videoChanged = true;
+          }
+
+          if (firstPlay) {
```

Comment:
you can just not check, and reassign `firstPlay = false` always. To be clearer, I mean replace

```javascript
if (firstPlay) {
  firstPlay = false;
}
```

with just

```javascript
firstPlay = false;
```